server/routes/users/index.ts

```diff
@@ -4,7 +4,34 @@ import userController from '../../controllers/userController';
 import validateRefreshJWT from '../../validation/validateRefreshJWT';
 import verifyUser from '../../validation/verifyUser';
 import wrap from '../../controllers/wrap';
+import { initManagers } from '../../db/initDB';
+import validateFile from '../../validation/validateFile';
 import express = require('express');
+import multer = require('multer');
+import path = require('path');
+
+const { userManager } = initManagers();
+
+const store = multer.diskStorage({
+	destination: path.resolve(__dirname, '../../../uploads/'),
+	filename: async function(req, file, cb) {
+		const fileObj = {
+			'image/png': '.png',
+			'image/jpeg': '.jpeg',
+			'image/jpg': '.jpg'
+		};
+		const refreshToken = req.cookies.refreshToken;
+		const username = await userManager.findRefreshToken(refreshToken);
+		cb(null, username + fileObj[file.mimetype]);
+	}
+});
+
+const upload = multer({
+	storage: store,
+	fileFilter: function(req, file, cb) {
+		validateFile(file, cb);
+	}
```

Comment:
You can adjust the function to use it directly and specify its type according to multer as well.

```suggestion
	fileFilter: validateFile;
```